src/YoutubeAtomPlayer.tsx

```diff
@@ -285,16 +284,25 @@ const createOnReadyListener =
 const createInstantiateImaManager =
     (
         uniqueId: string,
-        imaAdTagUrl: string,
         id: string,
         adContainerId: string,
+        adTargeting: AdTargeting | undefined,
         imaManager: React.MutableRefObject<ImaManager | undefined>,
         adsManager: React.MutableRefObject<google.ima.AdsManager | undefined>,
     ) =>
     (player: YT.Player) => {
+        const adTargetingEnabled = adTargeting && !adTargeting.disableAds;
+        const adUnit =
+            adTargetingEnabled && adTargeting.adUnit ? adTargeting.adUnit : '';
+        const customParams =
+            adTargetingEnabled && adTargeting.customParams
+                ? adTargeting.customParams
+                : {};
```

Comment:
The defensive logic here is not so nice and a consequence of `adTargeting` being optional.

I think we could tidy this and other sections by making `adTargeting` mandatory - which I think it could be. Will tackle this as a separate PR.